components/style/rule_tree/mod.rs

```diff
@@ -148,6 +158,13 @@ struct RuleNode {
 
     /// The next item in the rule tree free list, that starts on the root node.
     next_free: AtomicPtr<RuleNode>,
+
+    /// Number of RuleNodes we have added to the free list since the last GC.
+    /// (We don't update this if we rescue a RuleNode from the free list.  It's
+    /// just used as a heuristic to decide when to run GC.)
+    ///
+    /// Only used on the root RuleNode.
+    free_count: AtomicUsize,
```

Comment:
Seems unfortunate to pay this price in every node only for this, probably we should be able to reuse one of the `sibling` pointers for that, but that's fine as a followup I guess. Could you leave a comment here on that?